objects/bee103/script.lua

```diff
@@ -333,42 +334,40 @@ function getStage(o)
     return nil
 end
 ---- Fear Section
+function updateFearUI(params)
+    self.editButton({index = 1, label = params.fearPool})
+    UI.setAttribute("panelFearPool", "text", params.fearPool)
+    self.editButton({index = 2, label = params.generatedFear})
+    UI.setAttribute("panelFearGenerated", "text", params.generatedFear)
+end
 function addFear()
-    local fearPoolValue = self.getButtons()[2].label
-    local generatedFearValue = self.getButtons()[3].label
-    if tonumber(fearPoolValue) == 1 then
-        self.editButton({index = 1, label = generatedFearValue + 1})
-        Global.setVar("fearPool", generatedFearValue + 1)
-
-        self.editButton({index = 2, label = 0})
+    local fearPool = Global.getVar("fearPool")
```

Comment:
is there any particular reason why you're sourcing this value from a global variable rather than the button label? Personally i would imagine the current way of doing things is more efficient

Reply:
> Is there any particular reason why you're sourcing this value from a global variable rather than the button label? 

Given that the value saved and loaded is the global variable, it makes sense to me to treat that value as primary, and let the various places it is shown in the UI as derived values, 

> Personally i would imagine the current way of doing things is more efficient

I don't know enough of about the performance characteristics of lua in tabletop simulator to know. I'd guess  that given that this code is only run in response to user input, that either is fast enough.